06-data-carpentry.Rmd

```diff
@@ -555,12 +555,13 @@ The final thing to say about **dplyr** does not relate to the data but the synta
 This is fine when using R interactively. But when you'd like to use R non-interactively, code is generally more robust using standard evaluation: it minimises the chance of creating obscure scope-related bugs. Using standing evaluation also avoids having to declare global variables if you include the code in a package. For this reason most functions in **tidyr** and **dplyr** have two versions: one that uses NSE (the default) and another that uses standard evaluation which requires the variable names to be provided in quote marks. The standard evaluation versions of functions are denoted with the affix `_`. This is illustrated below with the `group_by()` and `summarise()` functions:
 
 ```{r}
+library(rlang) #for parse_quosure
```

Comment:
Thanks for this. I think the preceding paragraph and this code chunk would be better as follows. This explains what's going on and links to the relevant reference. If you update your PR in this direction I'll happily approve it. Many thanks for taking the time to put this together, an important improvement.

```
This is fine when using R interactively. But when you'd like to use R non-interactively, code is generally more robust using standard evaluation: it minimises the chance of creating obscure scope-related bugs. Using standing evaluation also avoids having to declare global variables if you include the code in a package. To overcome this the concept of 'tidy evaluation' was developed and implemented in the package **rlang** (part of the tidyverse) to provide functions to control when symbols are evaluated and when they are treated as text strings. Without going into detail, the code below demonstrates how tidy evaluation works (see the [`tidy-evaluation`](https://cran.r-project.org/web/packages/rlang/vignettes/tidy-evaluation.html) vignette for further information):
```

```{r}
library(rlang)
# 1: Default NSE function
group_by(cars, speed = cut(speed, c(0, 10, 100))) %>%
  summarise(mean_dist = mean(dist))
# 2: Evaluation from character string
group_by(cars, speed = !!parse_quosure("cut(speed, c(0, 10, 100))")) %>%
  summarise(mean_dist = !!parse_quosure("mean(dist)"))
# 3: Using !! to evaluate 'quosures' when appropriate
q1 = quo(cut(speed, c(0, 10, 100)))
q2 = quo(mean(dist))
group_by(cars, speed = !!q1) %>%
  summarise(mean_dist = !!q2)
```

Reply:
Note the output also looks better now:

```
# A tibble: 2 x 2
  speed    mean_dist
  <fct>        <dbl>
1 (0,10]        15.8
2 (10,100]      49.0
```

Reply:
Awesome. Yes, the text needs to be changed. Adding `speed` is much better. 
One thing I would like to suggest is to add [this vignette](https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html)  
 too since the tidyeval vignette is somewhat technical instead of lots of *hands-on* example.

Reply:
Could you update the PR accordingly? Otherwise I could update the text after it's been merged.

Reply:
Thanks, I changed. If something is wrong, you can merge and update it.